src/konnektor/network_tools/clustering/scaffold_clustering.py

```diff
@@ -0,0 +1,175 @@
+"""Clustering compounds based on scaffolds
+
+This clusterer attempts to cluster compounds based on their scaffolds.
+It is built on rdkit's rdScaffoldNetwork module.
+"""
+from collections import defaultdict
+import itertools
+import gufe
+import openfe
+from rdkit import Chem
+from rdkit.Chem import rdMolHash
+from rdkit.Chem.Scaffolds import rdScaffoldNetwork
+
+from ._abstract_clusterer import _AbstractClusterer
+
+
+class TwoDimensionalScaffoldClusterer(_AbstractClusterer):
```

Comment:
Hm, I changed the naming for clusterers in the past refactoring.
maybe we should call it:
```suggestion
class ScaffoldClusterer(_AbstractClusterer):
```